pages/desktop/user.py

```diff
@@ -165,8 +165,106 @@ def clear_field(self):
+
+
+class MyFavorites(Base):
+
+    _page_title = 'My Favorite Add-ons :: Collections :: Add-ons for Firefox'
+    _header_locator = (By.CSS_SELECTOR, "h2.collection > span")
+
+    _edit_collection_locator = (By.CLASS_NAME, 'edit')
+    _favorites_add_ons_elements_locator = (By.CSS_SELECTOR, '.separated-listing .item')
+    _add_on_name_locator = (By.CSS_SELECTOR, '.item h3 a')
+    _add_to_favorites_locator = (By.CSS_SELECTOR, 'a.favorite')
+
+    def __init__(self, base_url, selenium, user_name=None):
+        Base.__init__(self, base_url, selenium)
+        if (user_name is not None):
+            self.user_name = user_name.lower()
+            self.selenium.get("%s/collections/%s/favorites" % (self.base_url, self.user_name))
+        WebDriverWait(self.selenium, self.timeout).until(
+            lambda s: self.is_the_current_page)
+
+    def edit_collection(self):
+        self.selenium.find_element(*self._edit_collection_locator).click()
+        return EditCollection(self.base_url, self.selenium)
+
+    @property
+    def my_favorites_header_text(self):
+        return self.selenium.find_element(*self._header_locator).text
+
+    @property
+    def favorites_add_ons_count(self):
+        return len(self.selenium.find_elements(*self._favorites_add_ons_elements_locator))
+
+    def is_add_on_added_as_favorite(self, name):
+        is_added = self.selenium.find_element(*self._add_on_name_locator).text
+        return name in is_added
+
+    @property
+    def is_add_on_marked_as_favorite(self):
+        is_favorite = self.selenium.find_element(*self._add_to_favorites_locator).get_attribute('title')
+        return 'Remove from favorites' in is_favorite
+
+
+class EditCollection(Base):
+
+    _title_locator = (By.CSS_SELECTOR, 'header > h2')
+
+    _name_field_locator = (By.ID, 'addon-ac')
+    _search_list_locator = (By.CSS_SELECTOR, '#ui-id-1')
+    _add_to_collection_button_locator = (By.ID, 'addon-select')
+    _save_changes_locator = (By.CSS_SELECTOR, '%s input[type=submit]')
+
+    @property
+    def _page_title(self):
+        return "%s :: Add-ons for Firefox" % self.title
+
+    @property
+    def title(self):
+        return self.selenium.find_element(*self._title_locator).text
+
+    def __init__(self, base_url, selenium, collection_name=None, user_name=None):
+        # formats name for url
+        Base.__init__(self, base_url, selenium)
+        if (collection_name is not None):
+            self.collection_name = collection_name.replace(" ", "-")
+            self.collection_name = re.sub(r'[^_A-Za-z0-9\-]', '-', self.collection_name).lower()
+            self.collection_name = self.collection_name[:30]
+            self.selenium.get("%s/collections/%s/%s/edit" % (self.base_url, user_name, self.collection_name))
+        WebDriverWait(self.selenium, self.timeout).until(
+            lambda s: self.is_element_visible(*self._title_locator))
+
+    def click_description_tab(self):
```

Comment:
This is not used, we should avoid the temptation of adding methods until we have code that uses them.